datafusion/optimizer/src/optimizer.rs

```diff
@@ -243,7 +242,6 @@ impl Optimizer {
         let rules: Vec<Arc<dyn OptimizerRule + Sync + Send>> = vec![
             Arc::new(EliminateNestedUnion::new()),
             Arc::new(SimplifyExpressions::new()),
-            Arc::new(UnwrapCastInComparison::new()),
```

Comment:
this is great -- it will also reduce the number of times the entire plan tree gets walked/massaged